### ISSUE TITLE
"describe ingress" does not list service endpoints for ingresses not in kube-system

### ISSUE THREAD

**Issue by rifelpet** [kubernetes/kubectl#592]
**Kubernetes version** (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.1", GitCommit:"eec55b9ba98609a46fee712359c7b5b365bdd920", GitTreeState:"clean", BuildDate:"2018-12-13T19:44:19Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.7", GitCommit:"65ecaf0671341311ce6aea0edab46ee69f65d59e", GitTreeState:"clean", BuildDate:"2019-01-24T19:22:45Z", GoVersion:"go1.10.7", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: AWS
- **OS** (e.g. from /etc/os-release): N/A
- **Kernel** (e.g. `uname -a`): N/A
- **Install tools**: N/A
- **Others**:

**What happened**:
`kubectl describe ingress` on any ingress not in the kube-system namespace shows `<none>` endpoints for services that do indeed have endpoints.

```
Name:             test
Namespace:        test
Default backend:  default-http-backend:80 (<none>)
Rules:
  Host                       Path  Backends
  ----                       ----  --------
  example.com
                             /*   example:80 (<none>)
Annotations:
```

Ingresses that are in the kube-system namespace correctly have their endpoints listed:

```
Name:             test
Namespace:        kube-system
Address:
Default backend:  example:80 (X.X.X.X:80,Y.Y.Y.Y:80)
Rules:
  Host  Path  Backends
  ----  ----  --------
  *     *     example:80 (X.X.X.X:80,Y.Y.Y.Y:80)
Annotations:
```

**What you expected to happen**:

endpoints are listed for ingresses outside of the kube-system namespace. Something like:

```
Name:             test
Namespace:        test
Default backend:  default-http-backend:80 (X.X.X.X:80,Y.Y.Y.Y:80)
Rules:
  Host                       Path  Backends
  ----                       ----  --------
  example.com
                             /*   example:80 (X.X.X.X:80,Y.Y.Y.Y:80)
Annotations:
```

**How to reproduce it** (as minimally and precisely as possible):

1. Create ingresses in both the `kube-system` namespace and a different namespace. Both should point to valid services with "ready" endpoints.
2. Run `kubectl describe ingress` on both ingresses. The `kube-system` ingress will list the service's endpoints but the other namespace's ingress will not.

**Anything else we need to know**:

Running `kubectl -v 9 describe ingress` on the ingress in the other namespace shows some 404s for getting the service and its endpoints but incorrectly using the kube-system namespace rather than the same namespace in which the ingress is located. 

```
I0227 12:05:46.669052   43775 round_trippers.go:419] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.13.1 (darwin/amd64) kubernetes/eec55b9" -H "Authorization: Basic ******" 'https://$CLUSTER_NAME/api/v1/namespaces/kube-system/endpoints/example'
I0227 12:05:46.682031   43775 round_trippers.go:438] GET https://$CLUSTER_NAME/api/v1/namespaces/kube-system/endpoints/example 404 Not Found in 12 milliseconds
I0227 12:05:46.682071   43775 round_trippers.go:444] Response Headers:
I0227 12:05:46.682084   43775 round_trippers.go:447]     Content-Type: application/json
I0227 12:05:46.682092   43775 round_trippers.go:447]     Date: Wed, 27 Feb 2019 20:05:46 GMT
I0227 12:05:46.682099   43775 round_trippers.go:447]     Content-Length: 196
I0227 12:05:46.682178   43775 request.go:942] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"endpoints \"example\" not found","reason":"NotFound","details":{"name":"example","kind":"endpoints"},"code":404}
I0227 12:05:46.682351   43775 round_trippers.go:419] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.13.1 (darwin/amd64) kubernetes/eec55b9" -H "Authorization: Basic ************" 'https://$CLUSTER_NAME/api/v1/namespaces/kube-system/services/example'
I0227 12:05:46.695058   43775 round_trippers.go:438] GET https://$CLUSTER_NAME/api/v1/namespaces/kube-system/services/example 404 Not Found in 12 milliseconds
I0227 12:05:46.695089   43775 round_trippers.go:444] Response Headers:
I0227 12:05:46.695099   43775 round_trippers.go:447]     Content-Type: application/json
I0227 12:05:46.695105   43775 round_trippers.go:447]     Date: Wed, 27 Feb 2019 20:05:46 GMT
I0227 12:05:46.695110   43775 round_trippers.go:447]     Content-Length: 194
I0227 12:05:46.695176   43775 request.go:942] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"services \"example\" not found","reason":"NotFound","details":{"name":"example","kind":"services"},"code":404}
```

**Comment by yue9944882** [comment]
that's designed i think. when the ingress's backend is absent.

[https://github.com/kubernetes/kubernetes/blob/32333b0fc74bfa0352a23ecfae3772ea07c9755c/pkg/kubectl/describe/versioned/describe.go#L2297-L2305](https://github.com/kubernetes/kubernetes/blob/32333b0fc74bfa0352a23ecfae3772ea07c9755c/pkg/kubectl/describe/versioned/describe.go#L2297-L2305)

**Comment by rifelpet** [comment]
That would not explain why the non-default-backend has `<none>` though. I would expect my backend's service to have its endpoints listed just like they are for ingresses in the kube-system namespace.

**Comment by rifelpet** [comment]
It appears this was fixed in 1.14.0 by https://github.com/kubernetes/kubernetes/pull/72142

```
Name:             example
Namespace:        default
Address:
Default backend:  default-http-backend:80 (<none>)
Rules:
  Host              Path  Backends
  ----              ----  --------
  example.com
                    /*      example:80 (1.2.3.4:80)

```
